keras_cv/utils/preprocessing.py

```diff
@@ -323,3 +325,40 @@ def check_fill_mode_and_interpolation(fill_mode, interpolation):
             "Unknown `interpolation` {}. Only `nearest` and "
             "`bilinear` are supported.".format(interpolation)
         )
+
+
+def clip_bounding_box(bounding_boxes, bounding_box_format, images):
+    """clips bounding boxes to image boundaries"""
+    bounding_boxes = bounding_box.convert_format(
+        bounding_boxes,
+        source=bounding_box_format,
+        target="rel_xyxy",
+        images=images,
+    )
+    bounding_boxes, images, squeeze = bbox_formater._format_inputs(
+        bounding_boxes, images
+    )
+    x1, y1, x2, y2, rest = tf.split(
+        bounding_boxes, [1, 1, 1, 1, bounding_boxes.shape[-1] - 4], axis=-1
+    )
+    clipped_bounding_boxes = tf.concat(
+        [
+            tf.clip_by_value(x1, clip_value_min=0, clip_value_max=1),
+            tf.clip_by_value(y1, clip_value_min=0, clip_value_max=1),
+            tf.clip_by_value(x2, clip_value_min=0, clip_value_max=1),
+            tf.clip_by_value(y2, clip_value_min=0, clip_value_max=1),
+            rest,
+        ],
+        axis=-1,
+    )
+
+    clipped_bounding_boxes = bounding_box.convert_format(
+        clipped_bounding_boxes,
+        source="rel_xyxy",
+        target=bounding_box_format,
+        images=images,
+    )
+    clipped_bounding_boxes = bbox_formater._format_outputs(
+        clipped_bounding_boxes, squeeze
```

Comment:
if we need to use _format_inputs and _format_outputs here, lets just keep it localized to this file.  

We can duplicate these functions in this file.  There is always a balance between "don't repeat yourself" and re-using too many helper functions.  In this case, lets just duplicate them to this file.  In general, it is bad practice to import and use `_` methods from other files, as it makes it easy to accidentally break other components.

Reply:
yes, I was handling batching here before without these helper functions. https://github.com/keras-team/keras-cv/pull/555#discussion_r923890424.
the code overhead was lesser than adding these two helper functions again here just to use it once.